server/queue/queue-activemq/src/test/java/org/apache/james/queue/activemq/ActiveMQMailQueueFactoryTest.java

```diff
@@ -32,29 +32,74 @@
+        ActiveMQMailQueueFactory mailQueueFactory;
 
-    @BeforeEach
-    public void setUp(BrokerService brokerService) {
-        ConnectionFactory connectionFactory = new ActiveMQConnectionFactory("vm://localhost?create=false");
-        RawMailQueueItemDecoratorFactory mailQueueItemDecoratorFactory = new RawMailQueueItemDecoratorFactory();
-        NoopMetricFactory metricFactory = new NoopMetricFactory();
-        mailQueueFactory = new ActiveMQMailQueueFactory(connectionFactory, mailQueueItemDecoratorFactory, metricFactory);
-        mailQueueFactory.setUseJMX(false);
-    }
+        @BeforeEach
+        public void setUp(BrokerService brokerService) {
+            ConnectionFactory connectionFactory = new ActiveMQConnectionFactory("vm://localhost?create=false");
+            RawMailQueueItemDecoratorFactory mailQueueItemDecoratorFactory = new RawMailQueueItemDecoratorFactory();
+            NoopMetricFactory metricFactory = new NoopMetricFactory();
+            mailQueueFactory = new ActiveMQMailQueueFactory(connectionFactory, mailQueueItemDecoratorFactory, metricFactory);
+            mailQueueFactory.setUseJMX(false);
+            mailQueueFactory.setUseBlobMessages(false);
+        }
+
+        @AfterEach
+        public void tearDown() {
+            mailQueueFactory.destroy();
+        }
 
-    @AfterEach
-    public void tearDown() {
-        mailQueueFactory.destroy();
+        @Override
+        public MailQueueFactory<ManageableMailQueue> getMailQueueFactory() {
+            return mailQueueFactory;
+        }
     }
 
-    @Override
-    public MailQueueFactory<ManageableMailQueue> getMailQueueFactory() {
-        return mailQueueFactory;
+    @Nested
+    @ExtendWith(BrokerExtension.class)
+    public static class ActiveMQMailQueueFactoryBlobsTest implements MailQueueFactoryContract<ManageableMailQueue>, ManageableMailQueueFactoryContract {
+
+        static final String BASE_DIR = "file://target/james-test";
+
+        ActiveMQMailQueueFactory mailQueueFactory;
+        ActiveMQMailQueueBlobTest.MyFileSystem fileSystem;
+
+        @BeforeEach
+        public void setUp(BrokerService brokerService) {
+            fileSystem = new ActiveMQMailQueueBlobTest.MyFileSystem();
+            ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("vm://localhost?create=false");
+
+
+            FileSystemBlobTransferPolicy policy = new FileSystemBlobTransferPolicy();
+            policy.setFileSystem(fileSystem);
+            policy.setDefaultUploadUrl(BASE_DIR);
+            connectionFactory.setBlobTransferPolicy(policy);
+
+            RawMailQueueItemDecoratorFactory mailQueueItemDecoratorFactory = new RawMailQueueItemDecoratorFactory();
+            NoopMetricFactory metricFactory = new NoopMetricFactory();
+            mailQueueFactory = new ActiveMQMailQueueFactory(connectionFactory, mailQueueItemDecoratorFactory, metricFactory);
+            mailQueueFactory.setUseJMX(false);
```

Comment:
the problem comes from the fact we have a common (mandatory) setup that is not encapsulated in the object we use.
Why don't we put hardcode this configuration in ActiveMQMailQueueFactory ? Anyway, it's the only tested configuration so we should not be able to use another configuration, right ?

Reply:
No strong opinion here. I just intended to correct a failing test (where the blob policy was badly initialized).

If you are not happy with this, feel free to drop/rewrite this.